app/hotels/src/navigation/allHotels/AllHotelsNavigationScreen.js

```diff
@@ -39,12 +48,19 @@ export default class AllHotelsNavigationScreen extends React.Component<Props> {
     }
 
     return {
-      headerLeft: (
-        <HeaderBackButton
-          tintColor={Color.brand}
-          onPress={props.onBackClicked}
-        />
-      ),
+      headerLeft:
+        props.lastNavigationMode && props.lastNavigationMode === 'present' ? (
+          <HeaderButton left={true} onPress={closeNativeModal}>
+            <Text style={styles.closeButton}>
+              <Translation id="hotels.navigation.close.button" />
+            </Text>
+          </HeaderButton>
+        ) : (
+          <HeaderBackButton
+            tintColor={Color.brand}
+            onPress={props.onBackClicked}
+          />
+        ),
```

Comment:
We will need `HOC` in the future for all that static stuff (MMB will have it too).

Reply:
Maybe open an issue for that?

Reply:
Yeah let's postpone it till we need it.